Source/Core/Core/HW/GCMemcard/GCMemcard.cpp

```diff
@@ -1162,120 +1162,115 @@ u32 GCMemcard::ReadAnimRGBA8(u8 index, u32* buffer, u8* delays) const
   // Batten Kaitos has 0x65 as flag too. Everything but the first 3 bytes seems irrelevant.
   // Something similar happens with Wario Ware Inc. AnimSpeed
 
-  int bnrFormat = (flags & 3);
+  int bnr_format = (flags & 3);
 
-  u32 DataOffset = GetActiveDirectory().m_dir_entries[index].m_image_offset;
-  u32 DataBlock = GetActiveDirectory().m_dir_entries[index].m_first_block - MC_FST_BLOCKS;
-
-  if ((DataBlock > m_size_blocks) || (DataOffset == 0xFFFFFFFF))
-  {
+  u32 image_offset = GetActiveDirectory().m_dir_entries[index].m_image_offset;
+  if (image_offset == 0xFFFFFFFF)
     return 0;
-  }
-
-  u8* animData = (u8*)(m_data_blocks[DataBlock].m_block.data() + DataOffset);
 
-  switch (bnrFormat)
+  switch (bnr_format)
   {
   case 1:
-    animData += 96 * 32 + 2 * 256;  // image+palette
+    image_offset += 96 * 32 + 2 * 256;  // image+palette
     break;
   case 2:
-    animData += 96 * 32 * 2;
+    image_offset += 96 * 32 * 2;
     break;
   }
 
-  int fmts[8];
-  u8* data[8];
-  int frames = 0;
-
-  for (int i = 0; i < 8; i++)
+  // decode frame count and formats
+  const u32 pixels_per_frame = 32 * 32;
+  const u32 palette_indices = 256;
+  u32 data_length = 0;
+  int frame_count = 0;
+  std::array<int, 8> frame_formats;
+  std::array<u8, 8> frame_delays;
+  std::array<u32, 8> frame_offsets;
+  bool has_shared_palette = false;
+  for (int i = 0; i < 8; ++i)
```

Comment:
Something tells me the `8` here (and above in the `std::array` declarations) should be a constant...not immediately sure where it comes from though.

Reply:
It's the maximum amount of frames in an icon animation, which results from the 16 bit values storing the per-frame format and delay at 2 bits each. Should probably be a constant, yeah.